### PR TITLE
feat(mc-scripts): support custom views in the `config:sync` command

### PR DESCRIPTION
**Pull request by kark**
#### Summary

Screenshot 2023-11-06 at 09 55 48

### CODE REVIEW

File: packages/mc-scripts/src/utils/get-config-diff.spec.ts [hunk 1]
```diff
@@ -17,342 +21,538 @@ const createTestConfig = (customConfig = {}) => ({
...
-            uriPath: 'my-test-app/new',
-            defaultLabel: 'New Avenger',
-            permissions: ['ManageMyTestApp'],
-            labelAllLocales: [
-              {
-                locale: 'de',
-                value: 'Neu Avenger',
-              },
-            ],
+            name: 'viewMyTestView',
```
Comment (emmenko) [inline packages/mc-scripts/src/utils/get-config-diff.spec.ts:447]: Nit:

```suggestion
            name: 'view',
```

Same for the others

File: packages/mc-scripts/src/utils/graphql-requests.spec.ts [hunk 2]
```diff
@@ -21,197 +25,370 @@ afterAll(() => mockServer.close());
...
+                    'customers.customer_details.custom_fields',
+                  ],
+                  permissions: [
+                    {
+                      oAuthScopes: ['view_products', 'view_customers'],
+                      name: 'viewTestCustomView',
+                    },
+                    {
+                      oAuthScopes: [],
+                      name: 'manageTestCustomView',
```
Comment (emmenko) [inline packages/mc-scripts/src/utils/graphql-requests.spec.ts:254]:
```suggestion
                      name: 'view',
                    },
                    {
                      oAuthScopes: [],
                      name: 'manage',
```

Same for the others

File: packages/mc-scripts/src/utils/graphql-requests.ts [hunk 3]
```diff
@@ -146,6 +172,24 @@ const fetchCustomApplication = async ({
...
+  const customViewData = await requestWithTokenRetry<
+    TFetchCustomViewFromCliQuery,
+    TFetchCustomViewFromCliQueryVariables
+  >(FetchCustomViewFromCli, {
+    variables: { customViewId },
+    mcApiUrl,
+    headers: {
+      'x-custom-view-id': customViewId,
+      'x-graphql-target': GRAPHQL_TARGETS.SETTINGS_SERVICE,
+    },
```
Comment (emmenko) [inline packages/mc-scripts/src/utils/graphql-requests.ts:188]: We should we sending also the `x-application-id`

**Review by emmenko — COMMENTED** [review]
Great work! 🙌

File: packages/mc-scripts/src/utils/graphql-requests.ts (hunk 3 above)
Comment (kark) [reply packages/mc-scripts/src/utils/graphql-requests.ts]: I see, now it's passed in all query/mutation requests

File: packages/mc-scripts/src/utils/graphql-requests.spec.ts [hunk 4]
```diff
@@ -19,199 +23,377 @@ beforeAll(() =>
...
+  const applicationIdentifier = '__local:@@custom-view-host@@';
+  describe('fetch custom view data', () => {
+    beforeEach(() => {
+      mockServer.use(
+        graphql.query('FetchCustomViewFromCli', (_req, res, ctx) => {
+          return res(
+            ctx.data({
+              organizationExtensionForCustomView: {
+                organizationId: 'organization-id',
+                customView: {
```
Comment (CarlosCortizasCT) [inline packages/mc-scripts/src/utils/graphql-requests.spec.ts:236]: Would it be ok to use the `test-data` package for the `CustomView`s mocked data?

**Review by CarlosCortizasCT — APPROVED** [review]
Looks GREAT to me 💯 

Thanks Kacper 👏

**Review by Rhotimee — APPROVED** [review]
Great job @kark

File: packages/mc-scripts/src/utils/graphql-requests.spec.ts (hunk 4 above)
Comment (kark) [reply packages/mc-scripts/src/utils/graphql-requests.spec.ts]: Thank you, that's a great suggestion, updated in ad42bb017d285a44f57a2806fa048c8a8fe8b211

File: packages/mc-scripts/src/utils/get-config-diff.ts [hunk 5]
```diff
@@ -436,4 +491,90 @@ const getConfigDiff = (
...
+// Compute diff changes of the Custom View config.
+// NOTE: Only known keys are evaluated.
+export const getCustomViewConfigDiff = (
+  oldConfig: CustomViewData,
+  newConfig: CustomViewData
+) => {
+  const diff: string[] = [];
+
+  // Default Label
+  const nameDiff = getStringDiff({
```
[the remark this answers is not in the document]
Comment (kark) [reply packages/mc-scripts/src/utils/get-config-diff.ts]: Thanks, good catch 👍 Updated in ad42bb017d285a44f57a2806fa048c8a8fe8b211

**Pull request merged**